presto-cache/src/main/java/com/facebook/presto/cache/filemerge/FileMergeCachingInputStream.java

```diff
@@ -51,9 +55,16 @@ public void readFully(long position, byte[] buffer, int offset, int length)
     {
         FileReadRequest key = new FileReadRequest(path, position, length);
 
```

Comment:
I think we should not introduce cache quota here, we can leave this as it is. The only part which should worry about cacheQuota is cache writing methods(put).

Reply:
Agree. We should not have cache quota check on any read path. We only check cache quota on `put`.